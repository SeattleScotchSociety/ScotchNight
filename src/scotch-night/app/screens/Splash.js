import React, { Component } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import { NavigationActions } from 'react-navigation';
import FadeInView from '../components/FadeInView';

export default class Splash extends Component {
    constructor(props) {
        super(props);

        this._startApp = this._startApp.bind(this);
    }

    componentDidMount() {
        setTimeout(this._startApp, 4000);
    }

    _startApp = () => {
        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'MainApp' })]
        });

        this.props.navigation.dispatch(resetAction);
    };

    render() {
        return (
            <FadeInView
                style={{
                    flex: 1,
                    justifyContent: 'center',
                    alignItems: 'center'
                }}>
                <Image style={styles.image} source={require('../../assets/main.jpg')}>
                    <View style={{ flex: 1, justifyContent: 'space-around', alignItems: 'center' }}>
                        <Text style={styles.headerText}>Seattle Scotch Society</Text>
                        <Text style={styles.headerText}>Presents</Text>
                        <Text style={styles.headerText}>Scotch Night</Text>
                    </View>
                </Image>
            </FadeInView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        paddingTop: 20,
        backgroundColor: '#ecf0f1'
    },
    paragraph: {
        margin: 24,
        fontSize: 18,
        fontWeight: 'bold',
        textAlign: 'center',
        color: '#34495e'
    },
    image: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    headerText: {
        backgroundColor: 'transparent',
        color: 'white',
        fontSize: 36,
        fontWeight: '600'
    }
});
