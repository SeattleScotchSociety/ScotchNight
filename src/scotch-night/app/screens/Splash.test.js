import React from 'react';
import Splash from './Splash';

import renderer from 'react-test-renderer';
jest.mock('react-navigation', () => {
    return { NavigationActions: { reset: () => {}, navigate: () => {} } };
});

test('renders correctly', () => {
    const tree = renderer.create(<Splash navigation={{ dispatch: () => {} }} />).toJSON();
    expect(tree).toMatchSnapshot();
});
