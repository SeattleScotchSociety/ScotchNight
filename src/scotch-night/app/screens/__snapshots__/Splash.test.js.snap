// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 36,
          "fontWeight": "600",
        }
      }
    >
      Seattle
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 36,
          "fontWeight": "600",
        }
      }
    >
      Scotch
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 36,
          "fontWeight": "600",
        }
      }
    >
      Society
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Go To App
        </Text>
      </View>
    </View>
  </Image>
</View>
`;
